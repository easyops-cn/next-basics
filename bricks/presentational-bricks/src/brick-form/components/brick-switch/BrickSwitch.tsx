import React from "react";
import { Switch } from "antd";
import { SwitchProps } from "antd/lib/switch";
import { WrapperFormItem } from "../wrapper-form-item/WrapperFormItem";

export interface SwitchMap {
  truthy: any;
  falsy: any;
}

export interface LegacyBrickSwitchProps {
  configProps?: SwitchProps;
  onChange: Function;
  checked?: boolean | string;
  switchMap?: SwitchMap;
}

export function LegacyBrickSwitch({
  configProps,
  onChange,
  checked,
  switchMap
}: LegacyBrickSwitchProps): React.ReactElement {
  const checkValue: boolean = switchMap
    ? switchMap.truthy === checked
      ? true
      : false
    : !!checked;

  const handleOnChange = (checked: boolean, event: Event) => {
    const value = switchMap ? switchMap[checked ? "truthy" : "falsy"] : checked;
    onChange(value);
  };

  return (
    <Switch {...configProps} checked={checkValue} onChange={handleOnChange} />
  );
}

export const BrickSwitch = WrapperFormItem(LegacyBrickSwitch);
