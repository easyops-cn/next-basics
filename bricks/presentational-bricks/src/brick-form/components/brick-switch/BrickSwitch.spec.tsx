import React from "react";
import { shallow } from "enzyme";
import { LegacyBrickSwitch } from "./BrickSwitch";

describe("BrickSwitch", () => {
  it("should work", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<LegacyBrickSwitch onChange={onChange} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger change event", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<LegacyBrickSwitch onChange={onChange} />);
    wrapper.simulate("change");
    expect(onChange).toHaveBeenCalled();
  });
});
