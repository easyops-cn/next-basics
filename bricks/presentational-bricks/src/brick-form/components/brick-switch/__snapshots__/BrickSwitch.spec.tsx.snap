// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrickSwitch should work 1`] = `
<Switch
  checked={false}
  onChange={[Function]}
/>
`;
